### PR TITLE
Fix direction when changing floors

### PR DESCRIPTION
**Pull request by t0kenz**
... while standing on 3 stackable objects

### CODE REVIEW

**Comment by Kamenuvol** [comment]
But you can change floors going to <-- and it should be looking to the left

**Comment by t0kenz** [comment]
oh yeah, didn't think about it let me fix it :-)

File: src/game.cpp [hunk 1]
```diff
@@ -763,18 +763,20 @@ ReturnValue Game::internalMoveCreature(Creature* creature, Direction direction,
...
 	Position destPos = getNextPosition(direction, currentPos);
+	Player* player = creature->getPlayer();
 
 	bool diagonalMovement = (direction & DIRECTION_DIAGONAL_MASK) != 0;
-	if (creature->getPlayer() && !diagonalMovement) {
+	if (player && !diagonalMovement) {
 		//try go up
 		if (currentPos.z != 8 && creature->getTile()->hasHeight(3)) {
-			Tile* tmpTile = map.getTile(currentPos.x, currentPos.y, currentPos.getZ() - 1);
+			Tile* tmpTile = map.getTile(currentPos.x, currentPos.y, currentPos.z - 1);
```
Comment (ranisalt) [inline src/game.cpp]: Don't change this line as it's not related to the issue.

File: src/game.cpp [hunk 2]
```diff
@@ -763,18 +763,20 @@ ReturnValue Game::internalMoveCreature(Creature* creature, Direction direction,
...
 			if (tmpTile == nullptr || (tmpTile->getGround() == nullptr && !tmpTile->hasFlag(TILESTATE_BLOCKSOLID))) {
-				tmpTile = map.getTile(destPos.x, destPos.y, destPos.getZ() - 1);
+				tmpTile = map.getTile(destPos.x, destPos.y, destPos.z - 1);
```
Comment (ranisalt) [inline src/game.cpp]: Same as above.

**Comment by Kamenuvol** [comment]
nice name for a commit 😆

**Comment by ranisalt** [comment]
Thanks!

I felt like the boss now 🤣

**Pull request merged**